server/database/postgresql.js

```diff
@@ -0,0 +1,270 @@
+          .then(res => {
+            console.log('res insert i: ', res.rows[0]);
+            //console.log('client.release');
+            //client.release();
+
+          })
+          .catch(e => {
+            console.log('e : ', e.stack);
+            client.release();
+          })
+
+      }
+      //client.release();
+
+    })
+    .catch(e => {
+      console.log('e2 : ', e.stack);
+      client.release();
+    })
+
+}
+
+//delete response type and language
+const deletMasterData = (languageArray) => {
+  const deleteLanguages = 'Delete from Languages';
+  const deleteResponseType = 'Delete from Response_Time';
+  return pool.connect()
+    .then(client => {
+      return client.query(deleteLanguages)
+        .then(res => {
+          //console.log('Language deleted. Next, delete response type : ', res.rows[0]);
+          return client.query(deleteResponseType)
+        })
+        .then(res => {
+          client.release();
+          //console.log('Client releases : ', res.rows[0]);
+        })
+        .catch(e => {
+          client.release();
+          console.log(e.stack);
+        })
+    })
+
+}
+
+
+const insertResponseType = (responseTimeArr) => {
+  const nestedResponse = responseTimeArr.map(respTime => [respTime]);
+  const query = format('INSERT INTO RESPONSE_TIME (DESCRIPTION) VALUES %L returning id', nestedResponse);
+
+  return pool.connect()
+    .then(client => {
+      return client.query(query)
+        .then(res => {
+          //console.log('res insert responseTime : ', res.rows);
+          return res.rows;
+        })
+        .catch(e => {
+          client.release();
+          console.log(e.stack);
+        })
+    })
+
+}
+
+const insertLanguages = (languageArray) => {
+  const nestedArray = languageArray.map(language => [language]);
+  const query1 = format('INSERT INTO LANGUAGES (NAME) VALUES %L returning id', nestedArray);
+
+  return pool.connect()
+    .then(client => {
+      return client.query(query1)
+        .then(res => {
+          //console.log('res insert LANGUAGE : ', res.rows);
+          return res.rows;
+        })
+        .catch(e => {
+          client.release();
+          console.log(e.stack);
+        })
+    })
+
+}
+
+
+module.exports = {
+  insertOwner,
+  insertResponseType,
+  insertLanguages,
+  bulkInsertOwner,
+  bulkCopyCSV,
+  bulkCopyPGOwner,
+  tempTableToOwnersTable,
+  importCSVtoDB,
+  deletMasterData
+}
+
+
```

Comment:
Code here looks good. Is this file still WIP as well? I see a few libraries commented out that are used in the code. I also see a number of database inserts in your functions. Have you run into any blockers with this number of database queries?

Reply:
I need to comment out those code. I tried different approaches but those were taking time.  Inserts worked with those db queries but with 10 mil, it was running into memory heap error. Another approach where I copied data from csv to temp table then to main table, it was also timing out as well - which should have been fast but could be DB memory limit- need more analysis. Exporting to csv and importing was the fastest.